pkg/network/ebpf/c/kafka/kafka-helpers.h

```diff
@@ -0,0 +1,200 @@
+#ifndef __KAFKA_HELPERS_H
+#define __KAFKA_HELPERS_H
+
+#include "kafka-types.h"
+
+// Forward declaration
+static __always_inline bool try_parse_produce_request(kafka_transaction_t *kafka_transaction);
+static __always_inline bool try_parse_fetch_request(kafka_transaction_t *kafka_transaction);
+static __always_inline bool extract_and_set_first_topic_name(kafka_transaction_t *kafka_transaction);
+
+static __inline int32_t read_big_endian_int32(const char* buf) {
+    int32_t val;
+    bpf_probe_read_kernel(&val, 4, (void*)buf);
+    return bpf_ntohl(val);
+}
+
+static __inline int16_t read_big_endian_int16(const char* buf) {
+    int16_t val;
+    // De-Referencing buf causes misalignment verifier error for unknown reason, so using bpf_probe_read_kernel as a workaround
+    bpf_probe_read_kernel(&val, 2, (void*)buf);
+    return bpf_ntohs(val);
+}
+
+static __inline int32_t kafka_read_big_endian_int32(kafka_transaction_t *kafka_transaction) {
+    const int32_t value = read_big_endian_int32(kafka_transaction->request_fragment + kafka_transaction->current_offset_in_request_fragment);
+    kafka_transaction->current_offset_in_request_fragment += 4;
+    return value;
+}
+
+static __inline int16_t kafka_read_big_endian_int16(kafka_transaction_t *kafka_transaction) {
+    const int16_t value = read_big_endian_int16(kafka_transaction->request_fragment + kafka_transaction->current_offset_in_request_fragment);
+    kafka_transaction->current_offset_in_request_fragment += 2;
+    return value;
+}
+
+// Checking if the buffer represents kafka message
+static __always_inline bool try_parse_request_header(kafka_transaction_t *kafka_transaction) {
+    char *request_fragment = kafka_transaction->request_fragment;
+    if (request_fragment == NULL) {
+        return false;
+    }
+
+    const int32_t message_size = kafka_read_big_endian_int32(kafka_transaction);
+    log_debug("message_size: %d", message_size);
+    if (message_size <= 0) {
+        return false;
+    }
+
+    const int16_t request_api_key = kafka_read_big_endian_int16(kafka_transaction);
+    log_debug("request_api_key: %d", request_api_key);
+    if (request_api_key < 0 || request_api_key > KAFKA_MAX_VERSION) {
+        log_debug("request_api_key < 0 || request_api_key > KAFKA_MAX_VERSION");
+        return false;
+    }
+    kafka_transaction->request_api_key = request_api_key;
+
+    const int16_t request_api_version = kafka_read_big_endian_int16(kafka_transaction);
+    log_debug("request_api_version: %d", request_api_version);
+    if (request_api_version < 0 || request_api_version > KAFKA_MAX_API) {
```

Comment:
nit: I think the names of these constants are bit confusing :sweat_smile: 
* request_api_key > KAFKA_MAX_VERSION
* request_api_version > KAFKA_MAX_API

maybe 
* `KAFKA_MAX_VERSION => KAFKA_MAX_API_KEY`
* `KAFKA_MAX_API => KAFKA_MAX_API_VERSION`

WDYT?

Reply:
Removed `KAFKA_MAX_VERSION`.
Renamed `KAFKA_MAX_API` -> `KAFKA_MAX_API_VERSION`

Reply:
Actually it seems like I've confused the 2, fixed it now 😅